src/components/number_input.tsx

```diff
@@ -70,16 +65,15 @@ const NumberInput: FC<NumberInputProps> = ({
             {prefix && <InputGroup.Text>{prefix}</InputGroup.Text>}
             <FloatingLabel className="mb-3" controlId={id} label={label}>
                 <Form.Control
-                    ref={ref}
                     onChange={event => {
-                        setCursor(event.target.selectionStart) // Set the cursor position as the selectionStart
+                        setLocalValue(event.target.value)
                         handleChange(event.target.value)
                     }}
-                    type="text"
-                    inputMode="numeric"
-                    value={value != null ? value : ''}
+                    type="number"
+                    value={localValue}
```

Comment:
`value` should be set with `value` not `localValue`.

Reply:
with  value={value != null ? value : ''} works fine in chrome but did not work in safari. So changed to useState variable localValue. It works fine.

Reply:
Ok. I see what is going on. If the input string is something like ".", then parseFloat() returns `NaN` which is what is stored in the Store and in the DB.  The "." is lost.

So I see two options:
1) Store the number input value as a string in the Store and in the DB. 
2) Store the number input value in local state as a string and call `updateValue()` only when `parseFloat()` is not NaN.

My preference is for the second option. So, very close to the code that you have now.
We can talk about this more.

Reply:
Have modified the as per discussion with Charlie.  Updated in numberInput -   the call updateValue() is made only when parseFloat() is not NaN. Additional check was implement to show error, when number not in the expected range gets stored in the database e.g., negative numbers when min is 0.